processors/fda_dap/processor.py

```diff
@@ -117,27 +139,26 @@ def _ensure_fda_application_exists(self, fda_approval):
         }
         return base.writers.write_fda_application(self._conn, fda_application, 'fda')
 
-    def _generate_document_id(self, document, fda_approval):
-        # MD5 is 128 bits, as UUID, so we can use one in place of the other
-        name = ''.join([fda_approval['id'],
-                        document['name']])
-        return hashlib.md5(name.encode('utf-8')).hexdigest()
-
-    def _find_document(self, document_id):
-        return self._conn['database']['documents'].find_one(id=document_id)
+    def _find_document(self, document, fda_approval):
+        # FIXME: We're using "name" to identify if the document already exist
+        # in our DB. However, "name" is mutable, so it isn't a good candidate
+        # for a key. I couldn't find a better one, though.
```

Comment:
I see the issue. However, I have no solution either. I guess we'll see about it later on.